src/main/java/com/hillel/items_exchange/service/CategoryService.java

```diff
@@ -1,8 +1,78 @@
 package com.hillel.items_exchange.service;
 
+import com.hillel.items_exchange.dao.CategoryRepository;
+import com.hillel.items_exchange.dto.CategoriesNamesDto;
+import com.hillel.items_exchange.dto.CategoriesVo;
+import com.hillel.items_exchange.dto.CategoryVo;
+import com.hillel.items_exchange.mapper.CategoryMapper;
+import com.hillel.items_exchange.model.Category;
+import com.hillel.items_exchange.model.Subcategory;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class CategoryService {
+
+    private final CategoryRepository categoryRepository;
+    private final ModelMapper modelMapper;
+    private final CategoryMapper categoryMapper;
+
+    public CategoriesNamesDto findAllCategoryNames() {
+        CategoriesNamesDto categoriesNamesDto = new CategoriesNamesDto();
+        categoriesNamesDto.setCategoriesNames(Optional.ofNullable(categoryRepository.findAllCategoriesNames())
```

Comment:
Hibernate never return null instead of collection, so Optional is overabundant here - remove it.